app/src/main/java/ch/epfl/sdp/user/UserCache.kt

```diff
@@ -46,6 +46,22 @@ class UserCache {
         return true
     }
 
+    /**
+     * Check if a profile pic is cached, returns the corresponding bitmap if it exists or null otherwise
+     * @param context Context: The [Context] from which the call is operated
+     * @return Bitmap?: The Bitmap of the profile pic cached, or null if none is found
+     */
+    private fun retrieveProfilePic(context: Context): Bitmap? {
+        return try {
+            val imageStream = InputStreamReader(context.openFileInput(imageFilename))
+            imageStream.close()
+
+            BitmapFactory.decodeFile(context.filesDir.absolutePath + "/" + imageFilename)
+        } catch (e: Exception) {
+            null
+        }
```

Comment:
BitmapFactory.decodeFile does not raise any exception : https://developer.android.com/reference/android/graphics/BitmapFactory#decodeFile(java.lang.String)

If the is input stream is needed, tt might be good to only catch related IO exceptions